File: backend/Models/states/GameState.cs
using System.Text.Json;

namespace backend.Models.states;

public class GameState
{
    #region Properties
    public int TurnNumber { get; set; }
    public Guid RoomId { get; set; }
    public List<Player> PlayersList { get; set; }
    public List<List<HexTile>> HexTilesList { get; set; }
    
    #endregion

    
    public string Serialize()
    {
        return JsonSerializer.Serialize(this);
    }

    public GameState Deserialize(string data)
    {
       return JsonSerializer.Deserialize<GameState>(data)!;
    }

    public GameState(List<Player> playersList, List<List<HexTile>> hexTilesList, int turnNumber, Guid roomId)
    {
        PlayersList = playersList;
        HexTilesList = hexTilesList;
        TurnNumber = turnNumber;
        RoomId = roomId;
    }
}

File: frontend/civ_hex/lib/states/game_state.dart
import '../components/hex_tile.dart';
import '../models/player.dart';

class GameState {
  List<Player> players;
  List<HexTile> hexTiles;
  num turnNumber;

  GameState(
      {required this.players,
      required this.hexTiles,
      required this.turnNumber});
}
